overlord/configstate/configcore/hostname.go

```diff
@@ -41,7 +41,7 @@ func init() {
 // We are conservative here and follow hostname(7). The hostnamectl
 // binary is more liberal but let's err on the side of caution for
 // now.
-var validHostnameRegexp = regexp.MustCompile(`^[a-z0-9][a-z0-9-]{0,62}(\.[a-z0-9-]{1,63})*$`).MatchString
+var validHostnameRegexp = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-]{0,62}(\.[a-z0-9-]{1,63})*$`).MatchString
```

Comment:
the last letter could be upper case too AFAICT

```suggestion
var validHostnameRegexp = regexp.MustCompile(`^[a-zA-Z0-9][a-zA-Z0-9-]{0,62}(\.[a-zA-Z0-9-]{1,63})*$`).MatchString
```

Reply:
Samuele feedback was to do the same vaidations as systemd hostnamectl, i.e. accept whatever it accepts

Reply:
Silly question, is there a good description you are following here, or are you looking at hostnamectl source code? man hostname(7) is very sparse (and strict).

Reply:
Fwiw, this is what systemd is doing: https://github.com/systemd/systemd/blob/main/src/basic/hostname-util.c#L85